backend/corpus_asset_pipelines/summary_cubes/cell_count.py

```diff
@@ -0,0 +1,66 @@
+import logging
+
+import pandas as pd
+import tiledb
+
+from backend.wmg.data.schemas.corpus_schema import OBS_ARRAY_NAME
+from backend.wmg.data.schemas.cube_schema import cell_counts_schema
+from backend.wmg.data.snapshot import CELL_COUNTS_CUBE_NAME
+from backend.wmg.data.utils import create_empty_cube, log_func_runtime
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def extract(corpus_path: str) -> pd.DataFrame:
+    """
+    get obs data from integrated corpus
+    """
+    return tiledb.open(f"{corpus_path}/{OBS_ARRAY_NAME}")
+
+
+def transform(obs: pd.DataFrame) -> pd.DataFrame:
+    """
+    Create cell count cube data frame by grouping data in the
+    integrated corpus obs arrays on relevant features
+    """
+    df = (
+        obs.df[:]
+        .groupby(
+            by=[
+                "dataset_id",
+                "cell_type_ontology_term_id",
+                "tissue_ontology_term_id",
+                "assay_ontology_term_id",
+                "development_stage_ontology_term_id",
+                "disease_ontology_term_id",
+                "ethnicity_ontology_term_id",
+                "sex_ontology_term_id",
+                "organism_ontology_term_id",
+            ],
+            as_index=False,
+        )
+        .size()
+    ).rename(columns={"size": "n_cells"})
+    return df
+
+
+def load(corpus_path: str, df: pd.DataFrame) -> str:
+    """
+    write cell count cube to disk
+    """
+    uri = f"{corpus_path}/{CELL_COUNTS_CUBE_NAME}"
+    create_empty_cube(uri, cell_counts_schema)
+    tiledb.from_pandas(uri, df, mode="append")
+    return uri
+
+
+@log_func_runtime
+def create_cell_count_cube(corpus_path: str):
+    """
+    Create cell count cube and write to disk
+    """
+    obs = extract(corpus_path)
```

Comment:
Not a fan of the extract/transform/load naming convention here. I would rather use names that are more relevant to what each operation is effectively doing (for instance, the name `load` is overloaded and if you're not familiar with the ETL pattern, it will be confusing since it's effectively a `write` operation). @atolopko-czi your thoughts?

Reply:
I've personally disliked the "load" term for the same reason (I always think of it as loading data into memory, for whatever reason). But I figured sticking with ETL terminology would help communicate the pattern to anyone learning the code. If most of us dislike the "load" term (or other terms) I don't feel strongly if we use different terms, as long as the terms are consistent across all the (logical) steps in the pipeline.

Reply:
I completely agree  re `load` but I do think that there is real value is sticking with the etl pattern